python/python/lance/dataset.py

```diff
@@ -1843,6 +1851,20 @@ def validate(self):
         """
         self._ds.validate()
 
+    def migrate_manifest_paths_v2(self):
+        """
+        Migrate the manifest paths to the new format.
+
+        This will update the manifest to use the new v2 format for paths.
+
+        This function is idempotent, and can be run multiple times without
+        changing the state of the object store.
+
+        However, it should not be run while other concurrent operations are happening.
+        And it should also run until completion before resuming other operations.
```

Comment:
```suggestion
        DANGER: this should not be run while other concurrent operations are happening.
        And it should also run until completion before resuming other operations.
```

Very minor nit: can we use `DANGER` or `WARN` to draw a little more attention to this?